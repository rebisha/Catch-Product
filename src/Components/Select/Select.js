import React from "react";
import { func } from "prop-types";

import styles from "./select.scss";

const Select = ({ onChange }) => {
  return (
    <div className={styles.select}>
      <select
        name="sorting"
        id="sorting"
        className={styles.select__select}
        onChange={onChange}
      >
        <option value="">Sort by price</option>
        <option value="highest">Highest Price</option>
        <option value="lowest">Lowest Price</option>
      </select>
    </div>
  );
};

Select.propTypes = {
  onChange: func,
};

export default Select;
