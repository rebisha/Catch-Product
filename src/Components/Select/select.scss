@import "../../assets/styles/functions";

.select {
  display: flex;
  justify-content: flex-end;
  height: units(35px);

  &__select {
    width: auto;
  }
}
