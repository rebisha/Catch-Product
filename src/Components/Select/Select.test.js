import React from "react";
import renderer from "react-test-renderer";
// component
import Select from "./Select";

it("Should render correctly", () => {
  const wrapper = renderer.create(<Select />);
  expect(wrapper).toMatchSnapshot();
});
