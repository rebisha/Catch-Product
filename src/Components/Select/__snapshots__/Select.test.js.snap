// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="select"
>
  <select
    className="select__select"
    id="sorting"
    name="sorting"
  >
    <option
      value=""
    >
      Sort by price
    </option>
    <option
      value="highest"
    >
      Highest Price
    </option>
    <option
      value="lowest"
    >
      Lowest Price
    </option>
  </select>
</div>
`;
